algorithms/linfa-preprocessing/src/linear_scaling.rs

```diff
@@ -0,0 +1,571 @@
+            return Err(Error::FlippedMinMaxRange);
+        }
+        let mins = records.fold_axis(
+            Axis(0),
+            F::infinity(),
+            |&x, &prev| if x < prev { x } else { prev },
+        );
+        let mut scales =
+            records.fold_axis(
+                Axis(0),
+                F::neg_infinity(),
+                |&x, &prev| if x > prev { x } else { prev },
+            );
+        Zip::from(&mut scales).and(&mins).apply(|max, min| {
+            if abs_diff_eq!(*max - *min, F::zero()) {
+                // if feature is constant then don't scale
+                *max = F::one();
+            } else {
+                *max = F::one() / (*max - *min);
+            }
+        });
+        Ok(Self {
+            offsets: mins,
+            scales,
+            method: ScalingMethod::MinMax(min, max),
+        })
+    }
+
+    pub(crate) fn max_abs<D: Data<Elem = F>>(records: &ArrayBase<D, Ix2>) -> Result<Self> {
+        if records.dim().0 == 0 {
+            return Err(Error::NotEnoughSamples);
+        }
+        let scales = records.map_axis(Axis(0), |col| {
+            let norm_max = F::from(col.norm_max()).unwrap();
+            if abs_diff_eq!(norm_max, F::zero()) {
+                // if feature is constant at zero then don't scale
+                F::one()
+            } else {
+                F::one() / norm_max
+            }
+        });
+        let offsets = Array1::zeros(records.dim().1);
+        Ok(Self {
+            offsets,
+            scales,
+            method: ScalingMethod::MaxAbs,
+        })
+    }
+
+    /// Array of size `n_features` that contains the offset that will be subtracted to each feature
+    pub fn offsets(&self) -> &Array1<F> {
+        &self.offsets
+    }
+
+    /// Array of size `n_features` that contains the scale that will be applied to each feature
+    pub fn scales(&self) -> &Array1<F> {
+        &self.scales
+    }
+
+    /// Returns the method used for fitting. Useful for printing, since [ScalingMethod](enum.ScalingMethod.html) implements `Display`
+    pub fn method(&self) -> &ScalingMethod<F> {
+        &self.method
+    }
+}
+
+impl<F: Float> Transformer<Array2<F>, Array2<F>> for FittedLinearScaler<F> {
+    /// Scales an array of size (nsamples, nfeatures) according to the scaler's `offsets` and `scales`.
+    /// Panics if the shape of the input array is not compatible with the shape of the dataset used for fitting.
+    fn transform(&self, x: Array2<F>) -> Array2<F> {
+        if x.is_empty() {
+            return x;
+        }
+        let mut x = x;
+        Zip::from(x.gencolumns_mut())
+            .and(self.offsets())
+            .and(self.scales())
+            .apply(|mut col, &offset, &scale| {
+                if let ScalingMethod::Standard(false, true) = self.method {
+                    col.mapv_inplace(|el| (el - offset) * scale + offset);
+                } else {
+                    col.mapv_inplace(|el| (el - offset) * scale);
```

Comment:
what happens for `(false, false)`?

Reply:
Good catch 👍🏻 
Since the stds are set to 1 when `with_scaling` is set to false, I can just change the first condition to `(false, _)`. I added a test case for this